### PR TITLE
JDK17 Add Access.findBootstrapClassOrNull(name)

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Implemented `Access.findBootstrapClassOrNull(name)` via `VMAccess.findClassOrNull(name, ClassLoader.bootstrapClassLoader)`;
`JDK17` doesn't throw error when `bootstrapClassLoader.classLoaderValueMap` is initialized;
Moved `InvokeSpecialInsideInterface.java/DefineAnonClass.java` to Java `8` and `11+` specific folders to specify `jdk.internal.misc.Unsafe` for `Java 11+` (`JDK17` `sun.misc.Unsafe` removed `defineAnonymousClass()`).

With this PR, `-version` output:
```
openjdk version "17-internal" 2021-09-14
OpenJDK Runtime Environment (build 17-internal+0-adhoc.fengjcaibmcom.openj9-openjdk-jdk)
Eclipse OpenJ9 VM (build findbsc-8d1bf4139e, JRE 17 Mac OS X amd64-64-Bit Compressed References 20210515_000000 (JIT enabled, AOT enabled)
OpenJ9   - 8d1bf4139e
OMR      - 092787cd6
JCL      - a1284f52b71 based on jdk-17+22)
```

closes https://github.com/eclipse-openj9/openj9/issues/12695

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

**Comment by genie-openj9** [comment]
Can one of the admins verify this patch?

File: jcl/src/java.base/share/classes/java/lang/Access.java [hunk 1]
```diff
@@ -484,6 +484,11 @@ public String join(String prefix, String suffix, String delimiter, String[] elem
 		return String.join(prefix, suffix, delimiter, elements, size);
 	}
 	/*[ENDIF] OPENJDK_METHODHANDLES*/
+	
+	public Class<?> findBootstrapClassOrNull(String name) {
+		return VMAccess.findClassOrNull(name, ClassLoader.bootstrapClassLoader);
+	}
```
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/java/lang/Access.java:490]: The old signature should (eventually?) be removed:
```
public Class<?> findBootstrapClassOrNull(ClassLoader loader, String name)
```
as it has been removed from `JavaLangAccess`.

**Review by keithc-ca — COMMENTED** [review]
Please explain the motivation for all the test changes. I don't immediately see the connection.

**Comment by JasonFengJ9** [comment]
> Please explain the motivation for all the test changes. I don't immediately see the connection.

OpenJ9 test uses `sun.misc.Unsafe.defineAnonymousClass()` which has been removed in latest JDK17 update, and caused test compilation failure.

File: jcl/src/java.base/share/classes/java/lang/Access.java (hunk 1 above)
Comment (JasonFengJ9) [reply jcl/src/java.base/share/classes/java/lang/Access.java:492]: Removed it for JDK17.

**Comment by pshipton** [comment]
Is this still meant to be draft?

**Comment by JasonFengJ9** [comment]
> Is this still meant to be draft?

Still waiting for last test run to finish..

**Comment by pshipton** [comment]
jenkins test sanity win jdk8,jdk11,jdk16

**Comment by pshipton** [comment]
jenkins compile win jdknext depends ibmruntimes/openj9-openjdk-jdk#openj9-staging

File: jcl/src/java.base/share/classes/java/lang/Access.java [hunk 2]
```diff
@@ -484,6 +486,11 @@ public String join(String prefix, String suffix, String delimiter, String[] elem
...
```
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/java/lang/Access.java:492]: I think this belongs in the `else` branch of the conditional block starting on line 222.

**Comment by keithc-ca** [comment]
Are we ready to promote jdknext to the openj9-staging branch? If not, I think this breaks building against the (current) openj9 branch.

**Comment by pshipton** [comment]
We're not ready to promote, both variants should be kept working.

File: jcl/src/java.base/share/classes/java/lang/Access.java (hunk 2 above)
Comment (JasonFengJ9) [reply jcl/src/java.base/share/classes/java/lang/Access.java:492]: To keep both variants working, `findBootstrapClassOrNull(ClassLoader classLoader, String name)` can be removed after current `openj9-staging` contents promoted into `openj9` branch.
Updated PR.

**Review by pshipton — APPROVED** [review]

**Comment by keithc-ca** [comment]
Jenkins test sanity zlinux jdknext

**Review by llxia — APPROVED** [review]
The test change looks good to me.

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**